apiary.apib

```diff
@@ -203,6 +242,42 @@ Wrapper for a native Storage API call, that encrypts (image configuration encryp
             }
 
 
+## Configuration Migration [/{componentId}/configs/{configId}/migrate]
```

Comment:
Nešla by udělat migrace tak, že by se to převedlo automaticky při callu na `/encrypt` (nebo `/encrypt-new`)? Migrate by pak byla pouze UI vlastnost, API bych minimalizoval.

Reply:
popremyslim

Reply:
nevidim v tom vyhodu
- do encrypt-new bych to nedaval v zadnym pripade, ten call se proste nebude nikde pouzivat
- kdyz to dame do encrypt, tak to znamena, ze po dobu co to tam bude se budou presifrovat vsechny sifry (vytvorim konfiguraci, ulozim, zasifruje se, upravim, ulozim, presifruje se) coz je podivny

Reply:
nepřešifruje se, uloží se rovnou z první zašifrovaná tou nejaktuálnější šifrou. jakejkoliv update starší konfigurace to přešifruje (a tady souhlas, to bude v audit logu vypadat divně).